### PR TITLE
[SYCL][DOC] Update buffer_location spec to unblock USM host/shared allocations

### PR DESCRIPTION
**Pull request by tiwaria1**
Spec states that the runtime buffer_location property will not have an effect on shared/host allocations. This is incorrect as the property is meant to be used with host/shared allocations with the same behavior as device allocations.

https://github.com/intel/llvm/pull/6218
https://github.com/intel/llvm/pull/6220

### CODE REVIEW

File: sycl/doc/extensions/proposed/sycl_ext_intel_runtime_buffer_location.asciidoc [hunk 1]
```diff
@@ -77,16 +77,59 @@ supports.
 
 == Examples
 
+.Device Allocation example
```
Comment (steffenlarsen) [inline sycl/doc/extensions/proposed/sycl_ext_intel_runtime_buffer_location.asciidoc:80]:
```suggestion
.Device allocation example
```

File: sycl/doc/extensions/proposed/sycl_ext_intel_runtime_buffer_location.asciidoc [hunk 2]
```diff
@@ -77,16 +77,59 @@ supports.
...
+q.copy(device_array, host_array, kN).wait();
+
+...
+... = host_array[index];
+----
 
-sycl::queue q;
-q.parallel_for(sycl::range<1>(N), [=] (sycl::id<1> i){
-  data[i] *= 2;
+.Shared Allocation example
```
Comment (steffenlarsen) [inline sycl/doc/extensions/proposed/sycl_ext_intel_runtime_buffer_location.asciidoc:111]:
```suggestion
.Shared allocation example
```

**Review by steffenlarsen — COMMENTED** [review]
Just some small nitpicks, otherwise the changes LGTM.

**Review by steffenlarsen — APPROVED** [review]

**Comment by tiwaria1** [comment]
@steffenlarsen  Can we get this one merged?

**Pull request merged**